src/azure-cli/azure/cli/command_modules/role/tests/util.py

```diff
@@ -9,21 +9,21 @@
 ROLE_COMMAND_SLEEP_DURATION = 10
 
 
-def cmd_with_retry(self, *args, sleep_duration=ROLE_COMMAND_SLEEP_DURATION, max_retry=ROLE_COMMAND_MAX_RETRY, **kwargs):
-    """Retry self.cmd and checks until success."""
+def retry(func, sleep_duration=ROLE_COMMAND_SLEEP_DURATION, max_retry=ROLE_COMMAND_MAX_RETRY):
```

Comment:
This function is refactored to support retrying a code block (enclosed within a function).